bitsail-clients/bitsail-client-entry-flink/src/main/java/com/bytedance/bitsail/entry/flink/command/FlinkCommandArgs.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2022-2023 Bytedance Ltd. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.bytedance.bitsail.entry.flink.command;
+
+import com.bytedance.bitsail.client.api.command.CommandArgs;
+
+import com.beust.jcommander.Parameter;
+import lombok.Getter;
+import lombok.Setter;
+
+import static com.bytedance.bitsail.entry.flink.deployment.kubernetes.KubernetesDeploymentSupplier.KUBERNETES_CLUSTER_ID;
+
+/**
+ * Created 2022/8/5
+ */
+@Getter
+@Setter
+public class FlinkCommandArgs implements CommandArgs {
+  public static final String JOB_ID = "job-id";
+  private static final String JOBMANAGER_ARCHIVE_FS_DIR = "jobmanager.archive.fs.dir";
+  private static final String HISTORYSERVER_WEB_ADDRESS = "historyserver.web.address";
+  private static final String HISTORYSERVER_WEB_PORT = "historyserver.web.port";
+  private static final String HISTORYSERVER_ARCHIVE_FS_DIR = "historyserver.archive.fs.dir";
+  private static final String HISTORYSERVER_ARCHIVE_FS_REFRESH_INTERVAL = "historyserver.archive.fs.refresh-interval";
+
+
+  @Parameter(names = "--execution-mode",
+      required = true,
+      description = "Flink run action, eg: run or run-application")
+  private String executionMode;
+
+  @Parameter(names = {"--queue"},
+      description = "Yarn queue's name when you use yarn deployment mode.")
+  private String queue;
+
+  @Parameter(names = "--deployment-mode",
+      required = true,
+      description = "Specify the flink deployment mode, eg: yarn-per-job or yarn-session")
+  private String deploymentMode;
+
+  @Parameter(names = "--from-savepoint")
+  private String fromSavepoint;
+
+  @Parameter(names = "--skip-savepoint")
+  private boolean skipSavepoint;
+
+  @SuppressWarnings("checkstyle:MagicNumber")
+  @Parameter(names = "--priority",
+      description = "Specify the job's priority in resource manager, eg: yarn.")
+  private int priority = 5;
+
+  @Parameter(names = "--jm-address",
+      description = "Specify the job manager to use, eg: localhost:8081.")
+  private String jobManagerAddress;
+
+  @Parameter(names = "--" + KUBERNETES_CLUSTER_ID,
+          description = "The cluster-id of kubernetes")
+  private String kubernetesClusterId = "bitsail-job";
+
+  @Parameter(names = "--" + JOB_ID,
+          description = "The job id of running job")
+  private String jobId;
+
+  @Parameter(names = "--historyserver.enable", description = "Enable history server or not.")
```

Comment:
Not sure if you're referring to `jobId` or `historyServerEnable`?
For `jobId`: We use it when stopping a specific running job. We won't make it a default value because jobId is a unique id of each running job.

For `historyServerEnable`: It's not from Flink but BitSail's self-defined parameter. This parameter is a gate to prevent users from using default values that don't work in practice. Without this gate, we will get this ERROR log in JobManager:
```bash
2023-02-20 21:24:22,632 ERROR org.apache.flink.runtime.history.FsJobArchivist              [] - Failed to archive job.
org.apache.flink.core.fs.UnsupportedFileSystemSchemeException: Could not find a file system implementation for scheme 'hdfs'. The scheme is not directly supported by Flink and no Hadoop file system to support this scheme could be loaded. For a full list of supported file systems, please see https://ci.apache.org/projects/flink/flink-docs-stable/ops/filesystems/.
	at org.apache.flink.core.fs.FileSystem.getUnguardedFileSystem(FileSystem.java:531) ~[flink-dist_2.11-1.11.6.jar:1.11.6]
	at org.apache.flink.core.fs.FileSystem.get(FileSystem.java:408) ~[flink-dist_2.11-1.11.6.jar:1.11.6]
	at org.apache.flink.core.fs.Path.getFileSystem(Path.java:274) ~[flink-dist_2.11-1.11.6.jar:1.11.6]
	at org.apache.flink.runtime.history.FsJobArchivist.archiveJob(FsJobArchivist.java:71) ~[flink-dist_2.11-1.11.6.jar:1.11.6]
	at org.apache.flink.runtime.dispatcher.JsonResponseHistoryServerArchivist.lambda$archiveExecutionGraph$0(JsonResponseHistoryServerArchivist.java:57) ~[flink-dist_2.11-1.11.6.jar:1.11.6]
	at org.apache.flink.util.function.ThrowingRunnable.lambda$unchecked$0(ThrowingRunnable.java:49) ~[flink-dist_2.11-1.11.6.jar:1.11.6]
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1640) [?:1.8.0_342]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_342]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_342]
	at java.lang.Thread.run(Thread.java:750) [?:1.8.0_342]
Caused by: org.apache.flink.core.fs.UnsupportedFileSystemSchemeException: Hadoop is not in the classpath/dependencies.
	at org.apache.flink.core.fs.UnsupportedSchemeFactory.create(UnsupportedSchemeFactory.java:55) ~[flink-dist_2.11-1.11.6.jar:1.11.6]
	at org.apache.flink.core.fs.FileSystem.getUnguardedFileSystem(FileSystem.java:527) ~[flink-dist_2.11-1.11.6.jar:1.11.6]
```

Reply:
After offline sync-up, will use existing job-conf field as cluster-id, such as `<instance-id>-<job-name>`

Reply:
Will verify if historyserver params can be configured in runtime. If so, we can remove this logic and ask users to put it in part of flink properties instead.